### PR TITLE
Feature/build update

### PR DESCRIPTION
**Pull request by dabeshouse**
Aiming for a 2.6 release

Updates:
Java 8 -> 17
Jython 2.1 -> 2.7.3
JOGL 2.3.2 -> 2.4.0
VecMath 1.3.1 -> 1.5.2
Install4J 7 -> 10

Because of changes made to support the newer version of Jython, which includes renaming a few packages saved projects are not backward compatible and cannot be opened by earlier versions of Alice 2.

### CODE REVIEW

File: src/main/java/edu/cmu/cs/stage3/alice/core/summary/ModelSummary.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public String[] getMethodNames() {
 		if( getModel() != null ) {
 			String[] methodNames = new String[ getModel().responses.size() ];
 			for( int i=0; i<methodNames.length; i++ ) {
-				edu.cmu.cs.stage3.alice.core.response.UserDefinedResponse methodI = (edu.cmu.cs.stage3.alice.core.response.UserDefinedResponse)getModel().responses.get( i );
+				edu.cmu.cs.stage3.alice.core.responses.UserDefinedResponse methodI = (edu.cmu.cs.stage3.alice.core.responses.UserDefinedResponse)getModel().responses.get( i );
```
Comment (jennej) [inline src/main/java/edu/cmu/cs/stage3/alice/core/summary/ModelSummary.java:57]: the word response(s) no longer has any meaning to me

File: src/main/java/edu/cmu/cs/stage3/alice/authoringtool/AuthoringTool.java [hunk 2]
```diff
@@ -391,6 +391,20 @@ protected void handleCaughtThowable( Runnable source, Throwable t ) {
 		}
 	}
 
+	public static Class getClassForName(String classname) throws ClassNotFoundException {
+		return Class.forName(correctedClassName(classname));
+	}
+
+	private static String correctedClassName(String classname) {
```
Comment (jennej) [inline src/main/java/edu/cmu/cs/stage3/alice/authoringtool/AuthoringTool.java:402]: add a comment here like  // Added for Alice 2.6, this rename will break backwards compatibility for alice 2.5

for the unlikely but not impossible case that someone is digging through alice later and wondering how and when we got here

File: src/main/java/edu/cmu/cs/stage3/alice/scripting/jython/ScriptingFactory.java [hunk 3]
```diff
@@ -63,7 +66,7 @@ public ScriptingFactory() {
 			br.close(); // To fix resource leak
 			if( sb.length()>0 ) {
 				String script = sb.substring( 0, sb.length()-1 );
-				org.python.core.PyCode code = org.python.core.__builtin__.compile( script, pathname, "exec" );
+				org.python.core.PyCode code = Py.compile_flags(script, pathname, CompileMode.exec, Py.getCompilerFlags());
```
Comment (jennej) [inline src/main/java/edu/cmu/cs/stage3/alice/scripting/jython/ScriptingFactory.java:69]: love the python CompileMode changes

File: src/main/java/edu/cmu/cs/stage3/image/ImageIO.java [hunk 4]
```diff
@@ -62,10 +62,7 @@ public static String mapExtensionToCodecName( String extension ) {
 	}
 
 
-    public static java.awt.Image load( String codecName, java.io.InputStream inputStream ) throws java.io.IOException {
-        return load( codecName, inputStream, null );
```
Comment (jennej) [inline src/main/java/edu/cmu/cs/stage3/image/ImageIO.java:66]: so satisfying.

File: src/main/java/edu/cmu/cs/stage3/alice/authoringtool/JAlice.java [hunk 5]
```diff
@@ -67,11 +67,10 @@ public class JAlice {
         }*/
 		try {
 			if (AikMin.isMAC()){
-				com.apple.eawt.Application app = com.apple.eawt.Application.getApplication();
-				URL url = edu.cmu.cs.stage3.alice.authoringtool.JAlice.class.getResource("images/alice.png");
-				app.setDockIconImage( java.awt.Toolkit.getDefaultToolkit().getImage(url) );
+				Desktop app = Desktop.getDesktop();
```
Comment (jennej) [inline src/main/java/edu/cmu/cs/stage3/alice/authoringtool/JAlice.java:70]: I have no idea how the icons work, but I'm going to trust that this works :)

**Review by jennej — APPROVED** [review]
long live alice 2

File: src/main/java/edu/cmu/cs/stage3/alice/core/summary/ModelSummary.java (hunk 1 above)
Comment (dabeshouse) [reply src/main/java/edu/cmu/cs/stage3/alice/core/summary/ModelSummary.java:57]: I have no response

File: src/main/java/edu/cmu/cs/stage3/alice/authoringtool/JAlice.java (hunk 5 above)
Comment (dabeshouse) [reply src/main/java/edu/cmu/cs/stage3/alice/authoringtool/JAlice.java:70]: It seems to. :)

File: src/main/java/edu/cmu/cs/stage3/alice/authoringtool/AuthoringTool.java (hunk 2 above)
Comment (dabeshouse) [reply src/main/java/edu/cmu/cs/stage3/alice/authoringtool/AuthoringTool.java:402]: Good idea. Done.

File: src/main/java/edu/cmu/cs/stage3/alice/scripting/jython/ScriptingFactory.java (hunk 3 above)
Comment (dabeshouse) [reply src/main/java/edu/cmu/cs/stage3/alice/scripting/jython/ScriptingFactory.java:69]: It is clearer.

**Pull request merged**